src/mapper.rs

```diff
@@ -20,6 +20,8 @@ struct ClassMapping<'s> {
     original: &'s str,
     obfuscated: &'s str,
     members: HashMap<&'s str, Vec<MemberMapping<'s>>>,
+    // obfuscated_method_name -> [parameters_types -> Vec(original_method_name)]
+    members_with_params: HashMap<&'s str, HashMap<&'s str, Vec<MemberMapping<'s>>>>,
```

Comment:
maybe I’m just micro-optimizing here, but you could do this like so a bit better:

```rust
struct MemberMappings {
  all_mappings: Vec<MemberMapping>,
  mappings_by_params: HashMap<&str, Vec<usize>>,
}
```
Here, the `Vec<usize>` represents an index into `all_mappings`, so you don’t have to duplicate all the mappings, as they are quite heavy.

Reply:
ps. marking as resolved since we discussed this offline.